usvm-ts/src/test/kotlin/org/usvm/util/TSTestResolver.kt

```diff
@@ -1,12 +1,90 @@
 package org.usvm.util
 
+import org.jacodb.ets.base.EtsBooleanType
+import org.jacodb.ets.base.EtsLiteralType
+import org.jacodb.ets.base.EtsNeverType
+import org.jacodb.ets.base.EtsNullType
+import org.jacodb.ets.base.EtsNumberType
+import org.jacodb.ets.base.EtsPrimitiveType
+import org.jacodb.ets.base.EtsRefType
+import org.jacodb.ets.base.EtsStringType
+import org.jacodb.ets.base.EtsType
+import org.jacodb.ets.base.EtsUndefinedType
+import org.jacodb.ets.base.EtsVoidType
 import org.jacodb.ets.model.EtsMethod
+import org.usvm.TSObject
 import org.usvm.TSTest
+import org.usvm.UExpr
+import org.usvm.USort
+import org.usvm.extractBool
+import org.usvm.extractDouble
+import org.usvm.extractInt
+import org.usvm.memory.URegisterStackLValue
+import org.usvm.state.TSMethodResult
 import org.usvm.state.TSState
 
-@Suppress("UNUSED_PARAMETER")
 class TSTestResolver {
-    fun resolve(method: EtsMethod, state: TSState): TSTest {
-        TODO()
+
+    fun resolve(method: EtsMethod, state: TSState): TSTest = with(state.ctx) {
+        val model = state.models.first()
+        when (val methodResult = state.methodResult) {
+            is TSMethodResult.Success -> {
+                val returnValue = resolveExpr(model.eval(methodResult.value), method.returnType)
```

Comment:
lets avoid nested calculations in complicated places to make it easier for debug. For example, here it is easier if the expr is evaluated at the previous instruction.

```kotlin
val valueToResolve = model.eval(methodResult.value)
val returnValue = resolveExpr(valueToResolve, method.returnType)
```